icu4c/source/test/intltest/erarulestest.cpp

```diff
@@ -0,0 +1,130 @@
+    TESTCASE_AUTO_END;
+}
+
+void EraRulesTest::testAPIs() {
+    const char * calTypes[] = {
+        "gregorian",
+        //"iso8601",
+        "buddhist",
+        "chinese",
+        "coptic",
+        "dangi",
+        "ethiopic",
+        "ethiopic-amete-alem",
+        "hebrew",
+        "indian",
+        "islamic",
+        "islamic-civil",
+        "islamic-rgsa",
+        "islamic-tbla",
+        "islamic-umalqura",
+        "japanese",
+        "persian",
+        "roc",
+        //"unknown",
+        NULL
+    };
+
+    for (int32_t i = 0; calTypes[i] != NULL; i++) {
+        UErrorCode status = U_ZERO_ERROR;
+        const char *calId = calTypes[i];
+
+        LocalPointer<EraRules> rules1(EraRules::createInstance(calId, FALSE, status));
+        if (U_FAILURE(status)) {
+            errln(UnicodeString("Era rules for ") + calId + " is not available.");
+            continue;
+        }
+
+        LocalPointer<EraRules> rules2(EraRules::createInstance(calId, TRUE, status));
+        if (U_FAILURE(status)) {
+            errln(UnicodeString("Era rules for ") + calId + " (including tentative eras) is not available.");
+            continue;
+        }
+
+        int32_t numEras1 = rules1->getNumberOfEras();
+        if (numEras1 <= 0) {
+            errln(UnicodeString("Number of era rules for ") + calId + " is " + numEras1);
+        }
+
+        int32_t numEras2 = rules2->getNumberOfEras();
+        if (numEras2 < numEras1) {
+            errln(UnicodeString("Number of era including tentative eras is fewer than one without tentative eras in calendar: ")
+                    + calId);
+        }
+
+        LocalPointer<Calendar> cal(Calendar::createInstance("en", status));
+        if (U_FAILURE(status)) {
+            errln("Failed to create a Calendar instance.");
+            continue;
+        }
+        int32_t currentIdx = rules1->getCurrentEraIndex();
+        int32_t currentYear = cal->get(UCAL_YEAR, status);
+        int32_t idx = rules1->getEraIndex(
+                currentYear, cal->get(UCAL_MONTH, status) + 1,
+                cal->get(UCAL_DATE, status), status);
+        if (U_FAILURE(status)) {
+            errln("Error while getting index of era.");
+            continue;
+        }
+        if (idx != currentIdx) {
+            errln(UnicodeString("Current era index:") + currentIdx + " is different from era index of now:" + idx
+                    + " in calendar:" + calId);
+        }
+
+        int32_t eraStartYear = rules1->getStartYear(currentIdx, status);
+        if (U_FAILURE(status)) {
+            errln(UnicodeString("Failed to get the start year of era index: ") + currentIdx + " in calendar: " + calId);
+        }
+        if (currentYear < eraStartYear) {
+            errln(UnicodeString("Current era's start year is after the current year in calendar:") + calId);
+        }
+    }
+}
+
+void EraRulesTest::testJapanese() {
+    const int32_t HEISEI = 235; // ICU4C does not define constants for eras
```

Comment:
not ideal, but probably fine.

Reply:
ICU4J exposes some constants for modern eras as int. Heisei is currently 235. ICU4J document says it might be changed in new releases. Markus pointed out it's not a good idea to change the value of these constants.

In this release, I considered that we clearly state these constant values won't change in future releases. But at the same time, I really think we should drop era data before Meiji. This would change the era indexes one more time.

I think it's not too late to find out how to expose some Japanese era constants after we decide what to do with historic eras.